### PR TITLE
Demo node improvements

### PR DESCRIPTION
**Pull request by rdettai**
- Push the demo node image to DockerHub because it can be handy to play with the Demo node locally. 
- Change the import to use the http operator. To make that possible we also switch to the compressed file data instead of the tarballs.
- Remove the intermediate image layer with data files

### CODE REVIEW

File: demo-node/entrypoint.bash
Comment (dominiklohmann) [inline demo-node/entrypoint.bash:54]: Since we soon will show the definition on multiple lines, can you `s/ | /\\n| /g` for both this and the Suricata pipeline?

File: demo-node/entrypoint.bash
Comment (dominiklohmann) [inline demo-node/entrypoint.bash:47]: This one doesn't contain the `where #schema != suricata.stats` that the one from `import.bash` has.

File: demo-node/entrypoint.bash
Comment (rdettai) [reply demo-node/entrypoint.bash:54]: you mean replacing with 
```
tenzir 'load https https://storage.googleapis.com/tenzir-datasets/M57/zeek-all.log.zst \n| decompress zstd \n| read zeek-tsv \n| import'
```
?

File: demo-node/entrypoint.bash
Comment (rdettai) [reply demo-node/entrypoint.bash:47]: should we not get rid of that filter now that we support larger events?

Comment (dominiklohmann) [reply demo-node/entrypoint.bash:47]: I think we should keep it. The events are rather meaningless.

File: demo-node/entrypoint.bash
Comment (dominiklohmann) [reply demo-node/entrypoint.bash:54]: Yes, although I think you can remove the space before the newline as well.

Comment (rdettai) [reply demo-node/entrypoint.bash:54]: doesn't hurt much and it helps readability of the code so I would rather leave them

Comment (dominiklohmann) [reply demo-node/entrypoint.bash:54]: Fine, I don't feel too strongly about them.

**Comment by rdettai** [comment]
@dominiklohmann @tobim do you mind if I merge [RUN /demo-node/load.bash](https://github.com/tenzir/tenzir/blob/6980b5456a4268a813628a5e5672eddf2a56756f/Dockerfile#L226) with `RUN /demo-node/import.bash` and don't store the raw files locally anymore. Downloading them into a layer increases the image size by ~1GB which is actually quite a lot 😅. It wouldn't even involve that much more overhead at build time as the layer is invalidated anytime the base image/code changes anyway.

**Comment by dominiklohmann** [comment]
I think it'd be even better if you spin up a node, start the two pipelines using the API, and then wait for them to complete; that way, the demo node also shows the two pipelines as completed and you don't need to have them twice in the file.

**Comment by rdettai** [comment]
I'll just wait for Tobi's input as he's the original author and there might be some non-obvious reason for the extra layer

File: demo-node/load.bash
Comment (mavam) [inline demo-node/load.bash:5]: Why not this?

```bash
tenzir 'load https storage.googleapis.com/tenzir-datasets/M57/suricata.json.zst
       | decompress zstd
       | save stdout' > suricata.json
```

Comment (rdettai) [reply demo-node/load.bash:5]: I think we should rather get rid of the layer entirely and do the load and import within the same pipeline. Waiting for Tobi's confirmation on this.

Comment (rdettai) [reply demo-node/load.bash]: got verbal confirmation from Tobi

**Comment by tobim** [comment]
> I'll just wait for Tobi's input as he's the original author and there might be some non-obvious reason for the extra layer. In all cases I'm quite confident this spared 1GB might reduce the demo node startup of 10s or more.

The download size itself should be less than 1GB thanks to compression.
The original use case for it was to test loading the data into the image as part of the build.
I don't think we need this mode any more, so we can just as well get rid of that layer.

File: demo-node/entrypoint.bash
Comment (dominiklohmann) [inline demo-node/entrypoint.bash:63]: Would it make sense to change this to actually start the pipeline, and then to wait until it is completed? That way, the pipeline will show up as completed rather than as created.

Comment (rdettai) [reply demo-node/entrypoint.bash:63]: To me, the very best would even be not to preload the data at build time and let the user see the import of the Zeek and Suricata datasets happening live. This way the demo nodes would connect even faster because they would be more lightweight in every sense. But unfortunately if I remember correctly there were some OOM issues with that.

Comment (dominiklohmann) [reply demo-node/entrypoint.bash:63]: You can probe whether all pipelines are done like this:

```bash
curl -X POST http://127.0.0.1:5160/api/v0/pipeline/list |
  jq -e '.pipelines | map(.state) | all(. == "completed")'
```

Which you can use in a loop with some sleep to wait until both pipelines are done.

Afterwards, you can issue a `tenzir-ctl rebuild --all` to optimize the DB.

Unless I'm missing something, this would be the strictly better option.

> To me, the very best would even be not to preload the data at build time and let the user see the import of the Zeek and Suricata datasets happening live.

I'd also be in favor of that, but it would require expanding the tour a fair bit. It would, however, allow us to remove the special image entirely.

**Comment by rdettai** [comment]
I think this PR is derailing a bit. My original intent was just to push the demo image to DockerHub 😅 . I propose we merge this as is and open follow up items:
- add back a continuous pipeline (needs some product design first)
- preload the data using the pipeline API instead of the current combination of "CLI build time load + unstarted pipeline in the entrypoint "

**Comment by dominiklohmann** [comment]
Fine by me—please create the issues on the housekeeping board before merging.

**Comment by rdettai** [comment]
Done, still missing an approval though.

**Review by dominiklohmann — APPROVED** [review]

**Pull request merged**
